core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -543,6 +601,14 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalString(pipeline);
         out.writeBoolean(isRetry);
         out.writeLong(autoGeneratedTimestamp);
+        if (out.getVersion().after(Version.V_5_3_0_UNRELEASED)) { // TODO update to onOrAfter after backporting
+            if (contentType != null) {
```

Comment:
This is null when we serialize an empty index request. This happens with the way reindex works since the request is created but not fully populated